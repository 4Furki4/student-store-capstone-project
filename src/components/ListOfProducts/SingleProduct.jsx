/* eslint-disable @next/next/no-img-element */
import Link from "next/link";
import React from "react";

export default function SingleProduct({ product }) {
    return (
        <Link href={`products/${product.id}`}>
            <div
                className='w-full transform font-Poppins shadow-xl transition duration-500 xxs:my-[-2rem] xxs:scale-75 xxs:hover:scale-[80%]
                            sm:my-2 sm:w-2/4 sm:scale-90 sm:hover:scale-100 md:w-5/12 md:scale-95 lg:mx-3 lg:w-1/4 xl:w-1/5'
            >
                <div className='mb-1 overflow-hidden rounded-lg bg-white'>
                    <img
                        src={product.img}
                        alt='image'
                        className='h-64 w-full cursor-pointer'
                    />
                    <div className='px-4 py-6 text-center'>
                        <div className='flex justify-between '>
                            <div className=' text-body-color mb-2 text-lg leading-relaxed'>
                                {product.title}
                            </div>
                            <span className='custom-bold text-xl'>
                                â‚º{product.price}
                            </span>
                        </div>

                        <div className='flex items-baseline justify-between'>
                            <span
                                className='inline-block rounded-full bg-teal-200 px-2 text-xs 
                                           font-semibold uppercase tracking-wide text-teal-800'
                            >
                                {product.category}
                            </span>

                            <span className='text-xs'>
                                {product.location}/Turkey
                            </span>
                        </div>
                    </div>
                </div>
            </div>
        </Link>
    );
}
