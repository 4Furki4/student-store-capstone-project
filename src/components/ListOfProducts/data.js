const PRODUCTS = [
    {
        id: 1,
        title: "Apple iPad Pro",
        description:
            "With the power of Apple Pencil, iPad Pro transforms into an immersive drawing canvas and the world's best noteâ€‘taking device.",
        coverImage:
            "https://productimages.hepsiburada.net/s/32/375/10354042765362.jpg",
        price: 4000,
        category: "Electronics",
        location: "Izmir",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.085562,
            lng: 28.985118,
        },
    },

    {
        id: 2,
        title: "1994",
        description:
            "It's 1994, and the Republican Revolution is on its way. Democratic Party strategist, Melissa Carnes, a fierce, combative woman in the male-dominated world of high politics, heads to her home state of Ohio to resuscitate the floundering Senate campaign of Kyle Porter, a successful entrepreneur running for public office for the first time.",
        coverImage:
            "https://static01.nyt.com/images/2017/01/27/arts/27bookorwell/27bookorwell-superJumbo.jpg",
        price: 250,
        category: "Books",
        location: "Istanbul",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.01203,
            lng: 28.955573,
        },
    },

    {
        id: 3,
        title: "leather jacket",
        description:
            "Cumin spiced cheese that originates from the Leiden area in the Netherlands.",
        coverImage:
            "https://cdn.shopify.com/s/files/1/1126/3188/products/moto-leather-jacket-compressor_1024x1024.jpg?v=1567465461",
        price: 570,
        category: "Clothes",
        location: "Ankra",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.085562,
            lng: 28.985118,
        },
    },

    {
        id: 4,
        title: "Defacto-shirt",
        description:
            "Cumin spiced cheese that originates from the Leiden area in the Netherlands.",
        coverImage:
            "https://img.freepik.com/free-psd/white-t-shirt-with-silkscreen-mockup_125540-697.jpg?w=2000",
        price: 110,
        category: "Clothes",
        location: "Mersin",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 40.362242,
            lng: 28.888175,
        },
    },

    {
        id: 5,
        title: "The Secret",
        description:
            "The Secret is a 2006 self-help book by Rhonda Byrne, based on the earlier film of the same name. It is based on the belief of the pseudoscientific law of attraction, which claims that thoughts can change a person's life directly.",
        coverImage:
            "https://www.thesecret.tv/wp-content/uploads/2015/05/Daily-Teaching-Book.png",
        price: 220,
        category: "Books",
        location: "Adana",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.085562,
            lng: 28.985118,
        },
    },

    {
        id: 6,
        title: "Iphone 6 mini",
        description:
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
        coverImage:
            "https://www.notebookcheck-tr.com/uploads/tx_nbc2/Apple_new-ipad-pro-apple-pencil-pro-display_03182020.jpg",
        price: 2400,
        category: "Electronics",
        location: "Istanbul",
        type: "Product",
        condition: "Used",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 40.98871,
            lng: 29.097114,
        },
    },

    {
        id: 7,
        title: "Leyden Cheese",
        description:
            "Cumin spiced cheese that originates from the Leiden area in the Netherlands.",
        coverImage:
            "https://cdn11.bigcommerce.com/s-n2v2dr/images/stencil/1280x1280/products/270/548/IMG_24191__24870.1643319583.JPG?c=2",
        price: 200,
        category: "Food",
        location: "Hatay",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 36.275556,
            lng: 36.188218,
        },
    },

    {
        id: 8,
        title: "PS5",
        description:
            "Cumin spiced cheese that originates from the Leiden area in the Netherlands.",
        coverImage:
            "https://media.karousell.com/media/photos/products/2021/7/29/barely_used_ps5_digital_editio_1627530598_43aa2105_progressive",
        price: 14000,
        category: "Electronics",
        location: "Istanbul",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 40.88341,
            lng: 29.231715,
        },
    },

    {
        id: 9,
        title: "Desk",
        description:
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
        coverImage:
            "https://i.pinimg.com/736x/9e/44/2b/9e442b9ba70e94d710fcce6107058322--diy-office-desk-home-office.jpg",
        price: 1200,
        category: "Furniture",
        location: "Istanbul",
        type: "Product",
        condition: "Used",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.019283,
            lng: 28.853814,
        },
    },

    {
        id: 10,
        title: "Office chair",
        description:
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
        coverImage:
            "https://secure.img1-fg.wfcdn.com/im/96489692/resize-h600-w600%5Ecompr-r85/9470/94709448/Office+Chairs.jpg",
        price: 1250,
        category: "Furniture",
        location: "Istanbul",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.019283,
            lng: 28.853814,
        },
    },

    {
        id: 11,
        title: "Scooter",
        description:
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
        coverImage:
            "https://i0.wp.com/rayhaber.com/wp-content/uploads/2020/06/seatin-elektrikli-scooteri-satisa-sunuldu.jpg?resize=678%2C381&ssl=1",
        price: 1250,
        category: "Two-wheeler",
        location: "Istanbul",
        type: "Product",
        condition: "Used",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 41.026535,
            lng: 28.668298,
        },
    },

    {
        id: 12,
        title: "Another Office chair",
        description:
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
        coverImage:
            "https://i.pcmag.com/imagery/roundup-products/019VKPwoqCHKm8uIrSmRYsb..v1646338208.jpg",
        price: 1250,
        category: "Furniture",
        location: "Istanbul",
        type: "Product",
        condition: "New",
        seller: "DerSeller",
        buyer: "DerBuyer",
        geo: {
            lat: 40.978861,
            lng: 28.643563,
        },
    },
];

export default PRODUCTS;
