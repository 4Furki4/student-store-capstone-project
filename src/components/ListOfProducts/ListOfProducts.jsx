import React from "react";

import PRODUCTS from "./data";
import SingleProduct from "./SingleProduct";
export default function ListOfProducts() {
    return (
        <div className='container mx-auto flex flex-wrap justify-center lg:justify-start '>
            {PRODUCTS.map((product) => {
                return <SingleProduct product={product} key={product.id} />;
            })}
        </div>
    );
}
