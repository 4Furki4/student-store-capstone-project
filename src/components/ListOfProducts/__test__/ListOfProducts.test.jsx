import renderer from "react-test-renderer";

import ListOfProducts from "../ListOfProducts";

it("U R THE Boss Well done", () => {
    const tree = renderer.create(<ListOfProducts />).toJSON();
    expect(tree).toMatchSnapshot();
});
