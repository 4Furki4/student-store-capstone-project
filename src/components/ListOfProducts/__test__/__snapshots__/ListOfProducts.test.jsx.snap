// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`U R THE Boss Well done 1`] = `
<div
  className="container mx-auto flex flex-wrap justify-center gap-x-6 lg:justify-start"
>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://productimages.hepsiburada.net/s/32/375/10354042765362.jpg"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            Apple iPad Pro 
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            4000
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            electronics
          </span>
          <span
            className="text-xs"
          >
            izmir
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://static01.nyt.com/images/2017/01/27/arts/27bookorwell/27bookorwell-superJumbo.jpg"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            1994
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            250
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            books
          </span>
          <span
            className="text-xs"
          >
            istanbul
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://cdn.shopify.com/s/files/1/1126/3188/products/moto-leather-jacket-compressor_1024x1024.jpg?v=1567465461"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            leather jacket
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            570
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            clothes
          </span>
          <span
            className="text-xs"
          >
            ankra
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://img.freepik.com/free-psd/white-t-shirt-with-silkscreen-mockup_125540-697.jpg?w=2000"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            Defacto-shirt
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            110
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            clothes
          </span>
          <span
            className="text-xs"
          >
            mersin
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://productimages.hepsiburada.net/s/32/375/10354042765362.jpg"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            The Secret
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            220
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            books
          </span>
          <span
            className="text-xs"
          >
            adana
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="my-4 w-full transform font-Poppins shadow-xl transition duration-500 hover:scale-105 sm:w-2/3 md:w-5/12 lg:mx-3 lg:w-1/4 xl:w-1/5 "
  >
    <div
      className="mb-1 overflow-hidden rounded-lg bg-white"
    >
      <img
        alt="image"
        className="h-72 w-full cursor-pointer"
        src="https://www.notebookcheck-tr.com/uploads/tx_nbc2/Apple_new-ipad-pro-apple-pencil-pro-display_03182020.jpg"
      />
      <div
        className="p-8 text-center sm:p-9 md:p-7 xl:p-9"
      >
        <h3 />
        <div
          className="flex justify-between "
        >
          <div
            className=" text-body-color mb-2 text-xl leading-relaxed"
          >
            Iphone 6 mini
          </div>
          <span
            className="custom-bold text-2xl"
          >
            ₺
            2400
          </span>
        </div>
        <div
          className="flex items-baseline justify-between"
        >
          <span
            className="inline-block rounded-full bg-teal-200 px-2 text-xs font-semibold uppercase tracking-wide text-teal-800"
          >
            electronics
          </span>
          <span
            className="text-xs"
          >
            istanbul
            /Turkey
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
